local/compose/run.go

```diff
@@ -0,0 +1,141 @@
+	"github.com/docker/compose-cli/api/compose"
+	convert "github.com/docker/compose-cli/local/moby"
+	apitypes "github.com/docker/docker/api/types"
+	moby "github.com/docker/docker/pkg/stringid"
+)
+
+func (s *composeService) CreateOneOffContainer(ctx context.Context, project *types.Project, opts compose.RunOptions) (string, error) {
+	service, err := project.GetService(opts.Name)
+	if err != nil {
+		return "", err
+	}
+	if err = s.ensureImagesExists(ctx, project); err != nil {
+		return "", err
+	}
+	if err := s.ensureProjectNetworks(ctx, project); err != nil {
+		return "", err
+	}
+	if err := s.ensureProjectVolumes(ctx, project); err != nil {
+		return "", err
+	}
+	if err = s.ensureRequiredServices(ctx, project, service); err != nil {
+		return "", err
+	}
+
+	updateOneOffServiceConfig(&service, project.Name, opts)
+
+	err = s.createContainer(ctx, project, service, service.ContainerName, 1)
+	if err != nil {
+		return "", err
+	}
+
+	return service.ContainerName, err
+}
+
+func (s *composeService) Run(ctx context.Context, container string, detach bool) error {
+	if detach {
+		return s.apiClient.ContainerStart(ctx, container, apitypes.ContainerStartOptions{})
+	}
+
+	cnx, err := s.apiClient.ContainerAttach(ctx, container, apitypes.ContainerAttachOptions{
+		Stream: true,
+		Stdin:  true,
+		Stdout: true,
+		Stderr: true,
+		Logs:   true,
+	})
+	if err != nil {
+		return err
+	}
+	defer cnx.Close()
+
+	stdout := convert.ContainerStdout{HijackedResponse: cnx}
+	stdin := convert.ContainerStdin{HijackedResponse: cnx}
+
+	readChannel := make(chan error, 10)
+	writeChannel := make(chan error, 10)
+
+	go func() {
+		_, err := io.Copy(os.Stdout, cnx.Reader)
+		readChannel <- err
+	}()
+
+	go func() {
+		_, err := io.Copy(stdin, os.Stdin)
+		writeChannel <- err
+	}()
+
+	go func() {
+		<-ctx.Done()
+		stdout.Close() //nolint:errcheck
+		stdin.Close()  //nolint:errcheck
+	}()
+
+	// start container
+	err = s.apiClient.ContainerStart(ctx, container, apitypes.ContainerStartOptions{})
+	if err != nil {
+		return err
+	}
+
+	for {
+		select {
+		case err := <-readChannel:
+			return err
+		case err := <-writeChannel:
+			return err
+		}
+	}
+}
+
+func updateOneOffServiceConfig(service *types.ServiceConfig, projectName string, opts compose.RunOptions) {
+	if len(opts.Command) > 0 {
+		service.Command = opts.Command
+	}
+	slug := moby.GenerateRandomID()
+	service.Scale = 1
+	service.ContainerName = fmt.Sprintf("%s_%s_run_%s", projectName, service.Name, moby.TruncateID(slug))
+	service.Labels = types.Labels{
+		slugLabel:   slug,
+		oneoffLabel: "True",
+	}
+	service.Tty = true
+	service.StdinOpen = true
+}
+
+func (s *composeService) ensureRequiredServices(ctx context.Context, project *types.Project, service types.ServiceConfig) error {
+	err := InDependencyOrder(ctx, project, func(c context.Context, svc types.ServiceConfig) error {
+		if svc.Name != service.Name { // only start dependencies, not service to run one-off
```

Comment:
This will not just crate dependencies, but also services that have no relation with the one we want to run.
Will also try to create a service that depends on the target one and we skipped here

Reply:
dependencies are filtered on the cli side, same as when calling `compose up SERVICE SERVICE`.